Sa3ed Issues

AdhamHebeish opened the pull request:
### Changes ###
- Commit 9e8a4e9
Fixed the `follow_up_case` API access issue where the API was inaccessible to guest users.

- Commit fbeaa71
Fixed the `search_people` API query issue where two similar queries were called separately, taking up extra server storage.

- Commit 36051c8
Fixed significant bugs in JS files including duplicate country codes on multiple pages and the error text being visible when there is no error on the `usersearch` page.

- Commit 93e8b5e
Implemented the `get_lost_person_list` function on the `lost_person_search` page as the function was not implemented.

### Future Additions ###
- Added `WhitelistBypass` API file for future use when bypassing Frappe whitelist.
- Working on removing unused functions to free up server storage

Al0olo commented on sa3ed_app/www/lost_person_create_detail/index.js:
```diff
@@ -1,6 +1,20 @@
     data() {
+        frappe.call({
+            method:"sa3ed_app.api.WhitelistBypass.get_country_list",
+            callback: function(result) { 
```
In callbacks you should apply in kind of validation on the result to make sure it's valid for what you are going to do with the response in your function.

This should apply here and where applicable.

AdhamHebeish replied:
This issue should be solved by Commit 8dc005d

Al0olo commented on sa3ed_app/www/lost_person_search/index.html:
```diff
@@ -1,4 +1,10 @@
 {% extends base_template_path %}
 {% block content %}
-<div id="div_test_response"></div>
+<link rel="stylesheet" href="index.css" />
```
if you are using naming like: index.html & index.css this sheet will be auto applied (no need for the link). this should also be applied in other files

AdhamHebeish replied:
This issue should be solved by Commit 751b36a

Al0olo commented on sa3ed_app/www/lost_person_search/index.js:
```diff
@@ -1,23 +1,44 @@
+        method: "sa3ed_app.api.LostPersonEndPoints.get_lost_persons",
+        args: {args_obj:args},
+        callback: function(result) {
+            var container = document.getElementById("person_conatiner");
```
here to avoid any Run time issues with JS you need to check if the element is added to the DOM before using it.

AdhamHebeish replied:
This issue should be solved by Commit 8dc005d

AdhamHebeish commented:
### Changes ### 
- Commit 751b36a
Removed unused link and script tags as per request of @Al0olo
> if you are using naming like: index.html & index.css this sheet will be auto applied (no need for the link). this should also be applied in other files

- Commit 8dc005d
Fixed issue on `lost_person_page` where an element was used without checking if it existed and where `callback` functions had no validation in case of incorrect data as per request of @Al0olo
> here to avoid any Run time issues with JS you need to check if the element is added to the DOM before using it. 

> In callbacks you should apply in kind of validation on the result to make sure it's valid for what you are going to do with the response in your function.

Al0olo commented:
It's good from my side. If @AMRRAGHEB118 doesn't have any comments then it's good to go.

AMRRAGHEB118 commented on sa3ed_app/www/found_person_create_detail/index.js:
```diff
@@ -1,6 +1,23 @@
+            method:"sa3ed_app.api.WhitelistBypass.get_country_list",
+            callback: function(result) { 
+                message = result.message;
+                var select = document.getElementById("country_select");
```
Use `let` or `const` instead of `var` depending on the context.
[Var, Let, and Const – What's the Difference?](https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/)

AdhamHebeish replied:
This issue should be solved by Commit 2c03058

AMRRAGHEB118 commented on sa3ed_app/www/found_person_create_detail/index.js:
```diff
@@ -1,6 +1,23 @@
+        frappe.call({
+            method:"sa3ed_app.api.WhitelistBypass.get_country_list",
+            callback: function(result) { 
+                message = result.message;
```
Return `status_code` so you can verify that the API request was fulfilled.
Look to [ApiEndPoint.py](https://github.com/lavaloon-eg/sa3ed_app/blob/master/sa3ed_app/api/ApiEndPoint.py) and [FoundPersonEndPoints.py](https://github.com/lavaloon-eg/sa3ed_app/blob/33520d11731854093acc45a4bcd826efa7fc0bc7/sa3ed_app/api/FoundPersonEndPoints.py#L69)

AdhamHebeish replied:
This issue should be solved by Commit 2c03058

AdhamHebeish commented:
### Changes ###
- Commit 2c03058
Fixed issue where JS variables were all declared using `var` instead of `let` or `const` and where the `get_country_list` API did not return a `status_code` to verify if the request was fulfilled as per request of @AMRRAGHEB118 
>Use let or const instead of var depending on the context.

>Return status_code so you can verify that the API request was fulfilled.
